components/Dropdown.vue

```diff
@@ -0,0 +1,271 @@
+<script>
```

Comment:
Yes, these three components are not very applicable here

![image](https://user-images.githubusercontent.com/24985926/133035169-c2ad4d95-1ce6-4ad7-a57c-76d2ce6927fc.png)

Reply:
Ah i see now, ok (i really need to get this running). Though I don't think these should be external actions. The buttons in this row are all 'actions' and this requires a lot of custom code to support

Reply:
Ok, i've seen this running now. I don't think these need to be grouped on their own, we don't do this for any other resource type. That would mean the model's `<action>.external`s can be removed and this Dropdown. Any objections @nwmac ?

Reply:
@richard-cox   @nwmac  We may have more than three actions in the dropdown button, and it may look strange if there are more actions。

![image](https://user-images.githubusercontent.com/24985926/133358598-88cd7947-ce23-4239-8e8c-7aedd856c796.png)

Reply:
Ok, given the timing I think we'd like to suggest the following
- The white bar should be removed (this looks bad on light might and is the wrong colour in dark mode)
- The whole button should have the same hover state and also trigger the drop down on click
- The drop down position should be directly underneath the button with a minimum width of the button width
- The Dropdown component should be renamed `ActionDropdown`

Going forward / After PR merges
- The dashboard team need to tweak which actions appear in the drop down (remove `external` property & dynamically determined given space available)
- The dashboard team need to review how the button is visually designed

Reply:
I've made the suggested changes, and I've found that the changes are now more similar to the ButtonDropdown component.
The decision to use the component will be made after the dashboard team has decided on the visual design。

Reply:
Done